File: agent/library/src/main/java/org/matilda/di/MessagesModule.java
package org.matilda.di;

import dagger.Binds;
import dagger.Module;
import org.matilda.messages.*;

@Module(includes = {MessageHandlerModule.class})
public interface MessagesModule {
    @Binds
    MessageSender bindMessageSender(BinaryMessageSender messageReceiver);

    @Binds
    MessageReceiver bindMessageReceiver(BinaryMessageReceiver messageReceiver);

    @Binds
    MessageSerializer bindMessageSerializer(ProtobufMessageSerializer messageSerializer);
}

File: agent/library/src/main/java/org/matilda/di/MessagesModule.kt
package org.matilda.di

import dagger.Binds
import dagger.Module
import org.matilda.messages.*

@Module(includes = [MessageHandlerModule::class])
interface MessagesModule {
    @Binds
    fun bindMessageSender(messageReceiver: BinaryMessageSender): MessageSender

    @Binds
    fun bindMessageReceiver(messageReceiver: BinaryMessageReceiver): MessageReceiver

    @Binds
    fun bindMessageSerializer(messageSerializer: ProtobufMessageSerializer): MessageSerializer
}
